doc/feature/synapses.md

```diff
@@ -0,0 +1,189 @@
+Synapse support in Brain
+============
+
+The low-level data access for synapses provided by Brion needs a complementary
+API in Brain to support various common higher-level uses cases, like semantic
+synapse object access or full-circuit synapse density voxelization, to name the
+two extremes.
+
+## Requirements
+
+* Semantic synapse objects that give read-access to all properties
+* Array access which allows vectorization
+* Filtering of synapses for pathway use case (shared synapses of two GIDSets)
+* Lazy-loading to support out-of-core scenarios
+* User-transparent KVS caching
+
+## New dependencies
+
+None
+
+## API
+
+```cpp
+
+class Circuit
+{
+    SynapsesStream getAfferentSynapses( const GIDSet& gids ) const;
+    SynapsesStream getEfferentSynapses( const GIDSet& gids ) const;
+    SynapsesStream getSharedSynapses( const GIDSet& preGIDs,
+                                      const GIDSet& postGIDs ) const;
+
+    // convenience for upper getXXX( gids ).read( gids.size( )).get()
+    Synapses getAllAfferentSynapses( const GIDSet& gids ) const;
```

Comment:
I'd call it getAfferentSynapses and rename the other one above, which I'd consider less common, to getAfferentSynapseStream